src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheck.java

```diff
@@ -135,12 +135,21 @@
     public void visitToken(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         if (parent.getType() == TokenTypes.SLIST
-                && (!allowInSwitchCase
-                    || parent.getNumberOfChildren() != 1)) {
+                && (!allowInSwitchCase || hasSiblings(ast))) {
             log(ast, MSG_KEY_BLOCK_NESTED);
         }
     }
 
+    /**
+     * Checks whether the AST node has any siblings or not.
+     *
+     * @param ast node to examine
+     * @return {@code true} if the node has one or more siblings
+     */
+    private static boolean hasSiblings(DetailAST ast) {
+        return ast.getNextSibling() != null || ast.getParent().getFirstChild() != ast;
```

Comment:
The `ast` parent can't have zero children, as it already have one child: the `ast` itself.
This expression should be `ast.getParent().getNumberOfChildren()  > 1`.

The first version of this method was named `isLonelyChild(ast)`, but was inverted later to avoid negation in `(!allowInSwitchCase || !isLonelyChild(ast))`.

Reply:
Thanks, it is clearer to me now.

> ast.getParent().getFirstChild() != ast

Since we are looking for a lonely child, would it be better to swap the above with `ast.getPreviousSibling() != null`?

Reply:
> Since we are looking for a lonely child, would it be better to swap the above with `ast.getPreviousSibling() != null`? Then we are only looking at the next/previous sibling and not involving the parent.

That makes sense. The method is rewritten.